### PR TITLE
Document the required node version for the JS Bindings

### PR DESCRIPTION
**Pull request by bame-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]

File: docs/source/ops/requirements.rst [hunk 1]
```diff
@@ -26,3 +26,8 @@ Daml is tested using the following specific dependencies in default installation
 2. `Eclipse Adoptium <https://adoptium.net>`_ version 11 for Java.
 3. PostgreSQL 10.0
 4. Oracle Database 19.11
+
+Feature/Component System Requirements
+-------------------------------------
+
+1. :doc:`The JavaScript Client Libraries <../app-dev/bindings-ts/index>` is tested on Node 14.18.3. A version greater or equal to 14.18.3 is recommended.
```
Comment (stefanobaghino-da) [inline docs/source/ops/requirements.rst:33]: IIRC the TS bindings are a browser-side component. Why is Node required?

**Review by stefanobaghino-da — APPROVED** [review]

File: docs/source/ops/requirements.rst (hunk 1 above)
Comment (bame-da) [reply docs/source/ops/requirements.rst:33]: You tell me, I'm just reflecting a requirement from the GSG: https://github.com/digital-asset/daml/pull/13011/files#diff-7efe8ff9244239f5d653233b6f0e0f9f82cbf1dba65b84b780d9268e84b975d3R36

Comment (bame-da) [reply docs/source/ops/requirements.rst:33]: I guess we could argue that you don't technically have to use node at all to build a project with the bindings. But you do need the TypeScript compiler, right? Do we want to say anything about the recommended version of that?

Comment (cocreature) [reply docs/source/ops/requirements.rst:33]: You do need node in the getting started guide to run create-react-scripts & similar stuff and also to run the typescript compiler. You don’t need node to run the resulting js bundle in your browser.

Comment (stefanobaghino-da) [reply docs/source/ops/requirements.rst:33]: > You tell me, I'm just reflecting a requirement from the GSG: https://github.com/digital-asset/daml/pull/13011/files#diff-7efe8ff9244239f5d653233b6f0e0f9f82cbf1dba65b84b780d9268e84b975d3R36

The Getting Started Guide does use `npm`, so the requirement makes sense there. This doesn't mean one is forced to use it.

Comment (stefanobaghino-da) [reply docs/source/ops/requirements.rst:33]: > You do need node in the getting started guide to run create-react-scripts & similar stuff and also to run the typescript compiler. You don’t need node to run the resulting js bundle in your browser.

Are there requirements with regards to the TypeScript compiler version needed to compile generated code?

Comment (bame-da) [inline docs/source/ops/requirements.rst:33]:
```suggestion
1. :doc:`The JavaScript Client Libraries <../app-dev/bindings-ts/index>` are tested on Node 14.18.3. with typescript compiler 4.5.4. Versions greater or equal to these are recommended. 
```

Comment (bame-da) [reply docs/source/ops/requirements.rst:33]: @stefanobaghino-da I'd stick with this. Telling people a working set of tools is of value.

Comment (bame-da) [reply docs/source/ops/requirements.rst]: I added some. Could you confirm you are OK with how it is now?

Comment (stefanobaghino-da) [reply docs/source/ops/requirements.rst]: LgTm

**Pull request merged**